Lib/asyncore.py

```diff
@@ -181,11 +178,14 @@ def poll2(timeout=0.0, map=None):
                 pollster.register(fd, flags)
 
         r = pollster.poll(timeout)
+        ready = []
         for fd, flags in r:
-            obj = map.get(fd)
-            if obj is None:
-                continue
-            readwrite(obj, flags)
+            ready.append((map[fd], flags))
```

Comment:
This loop can be simplified to:

    ready = [(map[fd], flags) for fd, flags in r]